client/tinychain/math/operator.py

```diff
@@ -101,6 +101,99 @@ def backward(self):
         return Mul(self.args, Pow(self.subject, Sub(self.args, 1)))
 
 
+class Sin(Operator):
+    def __init__(self, subject):
+        Operator.__init__(self, subject, None)
+
+    def forward(self):
+        return Trigonometric.sin(self.subject)
+
+    def backward(self):
+        return Cos(self.subject)
+
+
+class Cos(Operator):
+    def __init__(self, subject):
+        Operator.__init__(self, subject, None)
+
+    def forward(self):
+        return Trigonometric.cos(self.subject)
+
+    def backward(self):
+        from ..collection.tensor import Dense
+        return Sub(Dense.zeros(self.subject.shape), Sin(self.subject))
```

Comment:
Could you use the `_derivative` helper function here? That way you won't have to assume that the `subject` is a `Dense` tensor (which is important since you've also implemented these operators for `Number` and `Sparse`)